### PR TITLE
Allow slider render for integer/numeric types. Fixes #1780

### LINKED ISSUE #1780: Allow slider render for integer types.
#### Description:

Allow slider render for integer types.
Currently, only string type is supported, integer is aways rendered as updown text box.

#### Benefits

The json schema's data type validation is performed by Helm, allowing that charts using slider can be safely deployed from Helm's command line too.

#### Example

```
        "size": {
          "type": "integer",
          "title": "Persistent Volume Size",
          "form": true,
          "render": "slider",
          "sliderMin": 1,
          "sliderMax": 100,
          "sliderUnit": "Gi",
        }
```

### PR DESCRIPTION
**Pull request by batiati**
Fixes #1780

- Feature implemented
- Tests updated
- Docs updated

### CODE REVIEW

**Comment by absoludity** [comment]
Looks good - thanks @batiati . I've just retriggered the one ci test which failed (timeout), but will leave the PR for @andresmgot to review.

File: dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/SliderParam.tsx [hunk 1]
```diff
@@ -101,7 +101,9 @@ class SliderParam extends React.Component<ISliderParamProps, ISliderParamState>
 
   private handleParamChange = (value: number) => {
     this.props.handleBasicFormParamChange(this.props.param)({
-      currentTarget: { value: `${value}${this.props.unit}` },
+      currentTarget: {
+        value: this.props.param.type === "string" ? `${value}${this.props.unit}` : value,
```
Comment (andresmgot) [inline dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/SliderParam.tsx:105]: the `unit` prop is now optional. Can you update the rest of the class to cover the case in which unit is not set? (lines 9 and 91)

Comment (batiati) [reply dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/SliderParam.tsx:105]: Please tell me if something went wrong when merging ... I'm not used to git, so I've spent more time learning about git than coding.

**Comment by andresmgot** [comment]
hi @batiati it seems that you added a commit with the review but you reverted it so there are no changes now. Is that what you intended? Now for example the changes in the docs are missing.

**Comment by batiati** [comment]
> s that you added a commit with the review but you reverted it so there are no changes now. Is that what you intended? Now f

Hi @andresmgot, I'm sorry, I'm still getting used to git. I was just trying to sync my fork and I made a mistake sending a rollback!

If you can ignore this revert, the [0c573f3](https://github.com/kubeapps/kubeapps/pull/1788/commits/0c573f32343e5786c8c4a633a0a6f5c072366949) has all the changes requested, plus the documentation updates.

Or, if you prefer I can send a new PR with all changes at once.
Thank you.

**Comment by andresmgot** [comment]
Hi @batiati, I (Github) cannot ignore a commit since it's already in the remote branch. You can either revert 75b9d70 so the state goes back to how it was in 0c573f3 or you can send a new PR with the real changes, whatever is easier for you.

**Pull request closed without merging**